src/views/ApartmentApp/index.js

```diff
@@ -186,20 +224,26 @@ export default class ApartApp extends Component {
       this.saveButtonAlertTimeout = setTimeout(() => {
         this.saveButtonAlertTimeout = null;
         this.setState({ saving: false });
-      }, 2000);
+      }, 6000);
     }
   }
 
-  handleCloseOkay = () => {
-    this.setState({ submitDialogOpen: false, errorDialogOpen: false, errorDialogText: null });
+  handleCloseSnackbar = (event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    this.setState({ snackbarOpen: false });
   };
 
-  handleCloseSnackbar = (event, reason) => {
+  handleCloseDialog = (event, reason) => {
     if (reason === 'clickaway') {
       return;
     }
+    this.handleCloseOkay();
+  };
 
-    this.setState({ snackbarOpen: false });
+  handleCloseOkay = () => {
+    this.setState({ submitDialogOpen: false, editDialogOpen: false });
```

Comment:
I see you removed the code that set the dialog text to null. Is this because it will always be set to something else before the dialog box opens? We do not need to worry about opening the dialog box without the text being changed?

Reply:
I am not sure which line you are referring to from the old code, 
In the latest version, the dialog box uses a constant text (it gets the text from the 'primaryApplicantAlertText' variable, for example), so the text if never changed.
I originally thought I would use one dialog for all alerts and edit the text, but the flaw with that design is the each alert situation needs to map the 'confirm' and/or 'close' button to a unique callback function for that situation. 
If dialog boxes are needed for other messages or confirmations in the future, it is best to create them as separate DialogBox components, which I why I created the new `AlertDialogBox` component

Reply:
I tried an approach of handleing the callback's behavior dynamically when it was setup to change the text, but that was bugging due to weird race conditions. I found that attempting to use a single dialog box dynamically led to a lot of unnecessary computational complexity compared to using multiple statically-defined dialog boxes. (which leads to better page response/performance in the web browser)

Reply:
Nice! The Alert Dialog Box component looks like a great solution!